python/ray/_private/usage/usage_lib.py

```diff
@@ -135,6 +142,42 @@ class UsageStatsEnabledness(Enum):
     ENABLED_BY_DEFAULT = auto()
 
 
+_library_usage_queue = queue.Queue()
+
+
+def record_library_usage(library: str):
+    _library_usage_queue.put(library)
+
+
+def _report_library_usage(library_usage_queue):
```

Comment:
This seems a bit overkill. How about we do a synchronous put, and just cache in a global var if it's already happened and ignore further calls for the given library?

Reply:
I don't think that's true at all. KV is used synchronously from plenty of libraries and this has never been an issue. I'd also much rather us start with something very simple before trying to optimize it.

Reply:
Yea, I also don't want premature optimization with complex code. I just think it's simple enough.

Anyway, I'll remove the thread.